File: src/types/member.ts
export interface Member {
  id: string;
  displayName: string;
  avatarEmoji: string | null;
  role: 'parent' | 'child';
  dateOfBirth: string;
  createdAt: string;
  familyId: string;
}

export interface Family {
  id: string;
  name: string;
  avatarEmoji: string | null;
  createdAt: string;
  memberCount: number;
  members: Member[];
}

export interface CreateFamilyRequest {
  name: string;
  avatarEmoji?: string;
}

export interface CreateMemberRequest {
  displayName: string;
  avatarEmoji?: string;
  role: 'parent' | 'child';
  dateOfBirth: string;
}

File: api/Models/Member.cs
namespace TheNumber.Api.Models;

public class Member
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string DisplayName { get; set; } = string.Empty;
    public string? AvatarEmoji { get; set; } = "🧒";
    public string Role { get; set; } = "child"; // "parent" | "child"
    public DateTime DateOfBirth { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid FamilyId { get; set; }
    public Family Family { get; set; } = null!;
}
